libcst/_nodes/expression.py

```diff
@@ -427,7 +427,7 @@ def _visit_and_replace_children(self, visitor: CSTVisitorT) -> "Integer":
 
     def _validate(self) -> None:
         super(Integer, self)._validate()
-        if not re.fullmatch(INTNUMBER_RE, self.value):
+        if not re.fullmatch(INTNUMBER_RE + "[Ll]?", self.value):
```

Comment:
I guess you decided that slightly lax validation on Py3 was preferable to introducing a separate `Py2Long` node type?

Reply:
Yes; another thing that's more lax than ideal here is that `tokenize.Intnumber` allows underscores (invalid on 2.7 and below) and `0o` prefix (invalid on 2.5 and below).  One relatively common practice in older code is `chmod(0177)` type calls, which isn't handled in this PR (it requires edits to the middle of the regex, at which point we should probably just reproduce it here).  I'm far more concerned with getting the default string type right in a subsequent PR than I am about stricter validation on ints.

Reply:
There was already some special-casing in the tokenizer, I expanded and made this more complete (see new `0030` series tests in the PR message)

Reply:
I called it Py2Integer though, to lump together long and octal py2-only int literals.